src/Magento/FunctionalTestingFramework/Console/RunTestCommand.php

```diff
@@ -104,4 +111,27 @@ function ($type, $buffer) use ($output) {
         }
         return $returnCode;
     }
+
+    /** Get an array of tests with resolved suite references from $testConfiguration
+     * eg: if test is referenced in a suite, it'll be stored in format suite:test
+     * @param string $testConfigurationJson
+     * @return array
+     */
+    private function resolveSuiteReferences($testConfigurationJson)
+    {
+        $testConfiguration = json_decode($testConfigurationJson, true);
+        $testsArray = $testConfiguration['tests'] ?? [];
+        $suitesArray = $testConfiguration['suites'] ?? [];
+        $testArrayBuilder = [];
+
+        foreach ($suitesArray as $suite => $tests) {
+            $testArrayBuilder = array_merge(
```

Comment:
See commend in BaseGenerateCommand, this can probably be simplified into one `foreach`